novawallet/Common/Storage/EntityToModel/ChainModelMapper.swift

```diff
@@ -183,6 +190,7 @@ final class ChainModelMapper {
         }
 
         entity.nodes = Set(nodeEntities) as NSSet
+        entity.selectedNode = try nodeMapping(model.selectedNode)
```

Comment:
The problem here is that ```nodeMapping``` makes a look up in the ```nodes``` list of the current chain and if it not there it creates a new entity which is just attached to the selectedNode and not in the nodes list. So, I would make a separate logic for the selectedNode that tries to find the existing node in the nodes list and if it is not there then raise an exception